bundles/org.openhab.binding.opengarage/src/main/java/org/openhab/binding/opengarage/internal/OpenGarageHandler.java

```diff
@@ -109,13 +100,6 @@ private void schedulePoll() {
         pollFuture = scheduler.scheduleWithFixedDelay(this::poll, 1, refreshInterval, TimeUnit.SECONDS);
     }
 
-    private synchronized void stopPoll() {
-        if (pollFuture != null && !pollFuture.isCancelled()) {
```

Comment:
Add `final Future<?> future = pollFuture;`, check for `null` and `isCancelled` on `future`, `future.cancel(true);` if necessary and set `pollFuture=null;` and the warning is gone.

Reply:
I don't completely understand this, but I think I did as you asked in a1f3facd2f